src/02-common-principles.md

```diff
@@ -400,108 +400,129 @@ datasets and non-compliant derivatives.
 
 ## The Inheritance Principle
 
-Any metadata file (such as `.json`, `.bvec` or `.tsv`) may be defined at any
-directory level, but no more than one applicable file may be defined at a given
-level (Example 1). The values from the top level are inherited by all lower
-levels unless they are overridden by a file at the lower level. For example,
-`sub-*_task-rest_bold.json` may be specified at the participant level, setting
-TR to a specific value. If one of the runs has a different TR than the one
-specified in that file, another `sub-*_task-rest_bold.json` file can be placed
-within that specific series directory specifying the TR for that specific run.
-There is no notion of "unsetting" a key/value pair.
-Once a key/value pair is set in a given level in the dataset, lower down in
-the hierarchy that key/value pair will always have some assigned value.
-Files for a particular participant can exist only at participant level directory,
-that is, `/dataset/sub-*[/ses-*]/sub-*_T1w.json`. Similarly, any file that is not
-specific to a participant is to be declared only at top level of dataset for example:
-`task-sist_bold.json` must be placed under `/dataset/task-sist_bold.json`
-
-Example 1: Two JSON files that are erroneously at the same level
+Any metadata file (such as `.json`, `.bvec` or `.tsv`) MAY be defined at any
+directory level. For any given data file, any metadata file at that directory
+level or higher that does not include any entities absent from the name of the
+data file and possesses the same suffix are applicable to that data file. Such
+files are loaded from the top of the directory hierarchy downwards, such that
+values from the top level are inherited by all data files at lower levels to
+which it is applicable unless overridden by a value for the same key present
+in another metadata file at a lower level (though it is RECOMMENDED to minimise
+the extent of such overrides). There is no notion of "unsetting" a
+key/value pair.
```

Comment:
```suggestion
Any metadata file (such as `.json`, `.bvec` or `.tsv`) MAY be defined at any directory level. 
For a given data file, any metadata file at that directory level or higher 
is applicable to that data file if:
- the metadata file and the data file possess the same suffix,
- the metadata file does not include any entities absent from the name of the data file. 
Such files are loaded from the top of the directory hierarchy downwards, 
such that values from the top level are inherited by all data files 
at lower levels to which it is applicable unless overridden 
by a value for the same key present in another metadata file at a lower level 
(though it is RECOMMENDED to minimise the extent of such overrides). 
There is no notion of "unsetting" a key/value pair.
```

Reply:
Yes, I'd contemplated bullet-pointing this myself; or potentially even enumerating so that datasets violating specific requirements / software code can be cross-referenced more precisely?

@VisLab: Good catch. There's going to need to be a hard split between JSON and other metadata files. For `.bvec` / `.bval` / `.tsv`, it I think needs to be simply selection of whichever applicable file is lowest in the directory tree; anything above it gets ignored, there's no reason to describe the contents of such as being "overwritten" by the content of files lower down. Only edge case that comes to mind would be `.tsv` files that happen to contain exactly the same columns (including their titles), but even then it's hard to justify choosing between a concatenation and a replacement. Are there any other metadata types that are worth being familiar with for the sake of getting the language / logic of the inheritance principle right?

Reply:
Good idea about using an ordered list and referring to those points in the example.

Reply:
So am I understanding that for JSON the resolution is top-down, with top (closest to the root) having precedence.
For the others it is bottom up, with the bottom (farthest from the root) having precedence?  

The notions of "lowest" and "highest" level in the tree isn't necessarily clear.  Does lowest mean farthest from the root? 

I honestly thought only JSON files got inherited in the sense that there could be multiple applicable ones for the same suffix.

Reply:
I think the only case of "official" of TSV inheritance we have around is having a single `task-foo_events.tsv` in the root of the directory in case all subjects / runs had the same design:

see from the BIDS examples

```bash
╰─⠠⠵ ls -l */*task*tsv
-rw-rw-r-- 1 remi remi  143 Nov 10 11:33 ds114/task-covertverbgeneration_events.tsv
-rw-rw-r-- 1 remi remi  280 Nov 10 11:33 ds114/task-fingerfootlips_events.tsv
-rw-rw-r-- 1 remi remi  143 Nov 10 11:33 ds114/task-overtverbgeneration_events.tsv
-rw-rw-r-- 1 remi remi  127 Nov 10 11:33 ds114/task-overtwordrepetition_events.tsv
-rw-rw-r-- 1 remi remi 1054 Nov 10 11:33 eeg_ds000117/task-facerecognition_channels.tsv
-rw-rw-r-- 1 remi remi 1807 Nov 10 08:42 synthetic/task-nback_events.tsv
```

Reply:
We probably need to expand the discussion of events inheritance.  Inherited `events.tsv files` are needed at several levels.  

For example, because BIDS forces multi-modality data to be separated into separate directories, simultaneously recorded MEG/EEG or fMRI/EEG will have the same event files for each run.  

In the following example from [hed-examples](https://github.com/hed-standard/hed-examples/tree/main/datasets/eeg_ds003654s_inheritance), EEG and MEG were recorded simultaneously, but separated into separate modality directories.  

```
ds_003654s:
       task-FacePerception_events.json
       sub-02:
               sub-002_task-FacePerception_events.json
               sub-002_task-FacePerception_run-1_events.tsv
               eeg:
                      sub-002_task-FacePerception_run-1_eeg.set 
               meg:
                      sub-002_task-FacePerception_run-1_meg.fif
   . . .
```

Right now we are assuming that multiple `events.json` files can be applicable to a given `events.tsv` files and that the keys are resolved from directory root down in the JSON files.  I think this is the right behavior for JSON files, but it needs to be clarified in the specification.

We have only made a single `events.tsv`  applicable to any given recording, but that `events.tsv` file could appear anywhere in the directory hierarchy.   

Events are a special case, and I think it might be feasible/useful to allow multiple events files at different levels to apply to a given run. The events are the union of the rows/join of columns.  This might be too complicated.  

The issue also comes up with derivatives.  Here the derivatives might have their own set of events. If the events from the raw data also apply do they need to always be copied into the events files? There was some discussion of this in the past, but there didn't seem to be a clear resolution or consensus on this when the discussion occurred.

Reply:
I also have a general question about inheritance.  Is the matching of entity components and their names case sensitive?

Reply:
> The notions of "lowest" and "highest" level in the tree isn't necessarily clear. Does lowest mean farthest from the root?

I always have to cross-reference elsewhere in the spec to remind myself which is which. Does anyone know if there's a robust reference-able definition somewhere? Or does this need to be defined upfront in the spec?

Reply:
> Is the matching of entity components and their names case sensitive?

I think the case sensitivity issues was solved in https://github.com/bids-standard/bids-specification/pull/858